### PR TITLE
implement spec for node population properties for vasculature, astroc…

### PR DESCRIPTION
**Pull request by mgeplf**
…yte, endfeet

ie:
    vasculatureFile
    vasculatureMesh
    endfeetMeshesFile
    microdomainsFile
    spineMorphologiesDir

### CODE REVIEW

File: python/bindings.cpp [hunk 1]
```diff
@@ -543,13 +543,28 @@ PYBIND11_MODULE(_libsonata, m) {
         m, "NodePopulationProperties", "Stores node population-specific network information")
         .def_readonly("spatial_segment_index_dir",
                       &NodePopulationProperties::spatialSegmentIndexDir,
-                      DOC_NODE_POPULATION_PROPERTIES(spatialSegmentIndexDir));
+                      DOC_NODE_POPULATION_PROPERTIES(spatialSegmentIndexDir))
+        .def_readonly("vasculature_file",
+                      &NodePopulationProperties::vasculatureFile,
+                      DOC_NODE_POPULATION_PROPERTIES(vasculatureFile))
+        .def_readonly("vasculature_mesh",
```
Comment (eleftherioszisis) [inline python/bindings.cpp:550]: This should be "vasculature_mesh_file"

Comment (mgeplf) [reply python/bindings.cpp:550]: yup, I'll change that; can you just use this name for the test?

Comment (eleftherioszisis) [reply python/bindings.cpp:550]: My bad. The test config is old. vasculature_mesh is the correct term, being also in the spec.

File: python/tests/test_config.py [hunk 2]
```diff
@@ -151,6 +151,33 @@ def test_biophysical_properties_raises(self):
...
+                "vasculature_mesh": "$BASE_DIR/vasculature_mesh.h5",
+            },
+            "networks": {
+                "nodes": [
+                    {
+                        "nodes_file": "$BASE_DIR/nodes1.h5",
+                        "populations": {
+                            "nodes-A": {
+                                "type": "vasculature",
+                                "vasculature_file": "$BASE_DIR/vasculature_file.h5",
```
Comment (eleftherioszisis) [inline python/tests/test_config.py:169]: The `vasculature_mesh` is below `vasculature_file` in ngv configs. The components are not usually used.

File: src/config.cpp [hunk 3]
```diff
@@ -655,29 +694,69 @@ class CircuitConfig::Parser
         result.biophysicalNeuronModelsDir = getJSONPath(components,
                                                         "biophysical_neuron_models_dir");
 
+        result.vasculatureFile = getOptionalJSONPath(components, "vasculature_file");
+        result.vasculatureMesh = getOptionalJSONPath(components, "vasculature_mesh");
+        result.endfeetMeshesFile = getOptionalJSONPath(components, "endfeetMeshesFile");
```
Comment (eleftherioszisis) [inline src/config.cpp:699]: Why is this in camel case, while the rest in snake case?

File: python/tests/test_config.py (hunk 2 above)
Comment (mgeplf) [reply python/tests/test_config.py:169]: yeah, I'm checking the resolution works, if it is in the config.

File: src/config.cpp (hunk 3 above)
Comment (mgeplf) [reply src/config.cpp]: tnx, fixed

**Review by eleftherioszisis — APPROVED** [review]

**Pull request merged**